client/option_advanced.go

```diff
@@ -206,6 +206,33 @@ func WithBoundHandler(h remote.BoundHandler) Option {
 	return Option{F: func(o *client.Options, di *utils.Slice) {
 		di.Push(fmt.Sprintf("AddBoundHandler(%T)", h))
 
-		o.RemoteOpt.AppendBoundHandler(h)
+		exist := false
+		switch h.(type) {
+		case remote.InboundHandler:
+			for _, inboundHandler := range o.RemoteOpt.Inbounds {
+				if inboundHandler == h.(remote.InboundHandler) {
```

Comment:
`switch handler := h.(type)`
then just use handler instead of h.(remote.InboundHandler)